internal/pkg/service/stream/api/service/aggregation.go

```diff
@@ -31,29 +29,40 @@ func (s *service) AggregateSources(ctx context.Context, d dependencies.BranchReq
 }
 
 func (s *service) addSinksToAggregationResponse(ctx context.Context, d dependencies.BranchRequestScope, response *stream.AggregatedSourcesResult) error {
-	sourceKeys := make(map[key.SourceKey]int)
+	// Collect source IDs
+	var sourceKeys []key.SourceKey
+	sourceKeyToIndex := make(map[key.SourceKey]int)
 	for i, source := range response.Sources {
 		sourceKey := key.SourceKey{
 			BranchKey: d.BranchKey(),
 			SourceID:  source.SourceID,
 		}
-		sourceKeys[sourceKey] = i
+		sourceKeys = append(sourceKeys, sourceKey)
+		sourceKeyToIndex[sourceKey] = i
 	}
 
-	sourcesWithSinks, err := d.AggregationRepository().GetSourcesWithSinksAndStatistics(ctx, maps.Keys(sourceKeys))
+	// Get sinks for all the sources
+	sourcesWithSinks, err := d.AggregationRepository().SourcesWithSinksAndStatistics(ctx, sourceKeys)
```

Comment:
Why can't we work with the map? Do we have to preserve order of input `response.Sources`?

Reply:
It looks better.